lib/utils/test/bpkg_error.bats

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env bats
+
+load "${PROJECT_ROOT}/test/test_setup.bash"
+
+# Load file under test
+load "$(dirname "${BATS_TEST_DIRNAME}")/utils.sh"
+
+@test "bpkg_error should call bpkg_message for a red error when called" {
+    # arrange
+    bpkg_message() { echo -n "$@" >&9; }
+
+    # act
+    bpkg_error
+
+    # assert
+    assert_equal "$(getMockContent)" 'red error'
```

Comment:
why not asserting on `bkpg_error` output directly? Like:

	assert_equal $(bkpg_error) "error:"  # no param
	assert_equal $(bkpg_error "foo bar") "error: foo bar"  # single param
	assert_equal $(bkpg_error "foo" "bar") "error: foo bar"  # multiple params

Probably need to redirect `bkpg_error` output

Reply:
Because `bpkg_error` does not output anything itself. 

It calls another function to actually output anything. As this is a unit-test (and the function called is not a native Bash function), the function used is mocked and the test only validates that the mocked function is called correctly.

Otherwise the test for `bpkg_error` could be broken by coding mistakes in either `bpkg_message` or `bpkg-term` (which it uses downstream).